boards/thingy52/Makefile.include

```diff
@@ -1,7 +1,6 @@
-# for this board, we are using Segger's RTT as default terminal interface
-USEMODULE += stdio_rtt
-TERMPROG = $(RIOTTOOLS)/jlink/jlink.sh
-TERMFLAGS = term-rtt
+# use JLink Segger RTT by default
+RIOT_TERMINAL ?= jlink
```

Comment:
I think this would make sense, it would replicate what is done in `Makefile.dep` in a way it would work if `stdio_ethos`, `stdio_uart` is included.

```suggestion
# HACK: replicate dependency resolution in Makefile.dep
ifeq (,$(filter stdio_%,$(DISABLE_MODULE)$(USEMODULE)))
  RIOT_TERMINAL ?= jlink
endif
```

Reply:
I don't think so. 'stdio-%' modules used/disabled are known only after makefile.dep has been parsed. So at his stage both variables are probably empty, since makefile.include is parsed before (for the moment). Maybe I'm missing something though.

Reply:
Unless:

a) passed my the comand line
b) disabled in the application makefile
c) required in the application makefile

Reply:
And for any of this cases it is required for `stdio_something` to be included as a module. In our code base it will not work for cases where `USEMODULE += stdio_ethos` is in an application `Makefile.dep`. So it is a HACK, and that is why I would have that comment, maybe extend it"

```
# HACK: replicate dependency resolution in Makefile.dep, only works of `USEMDOULE` or `DEFAULT_MODULE` is set by the command line or in the application Makefile.
```

It will still make the usage more transparent for most cases. Or maybe include `Makefile.dep` here and tag is as HACK, for me this would be OK, since we are moving towards including `Makefile.include` after `Makefile.dep` in which case this hack could be removed. What do you think?

Reply:
I applied you proposal for thingy52 and ruuvitag because they can use stdio over UART so that makes sense in this case. For hamilton, since there's no UART configured, I think we can keep the base `RIOT_TERMINAL ?= jlink` (maybe even without the `?`).

Reply:
> I applied you proposal for thingy52 and ruuvitag because they can use stdio over UART so that makes sense in this case. For hamilton, since there's no UART configured, I think we can keep the base RIOT_TERMINAL ?= jlink (maybe even without the ?).

I would keep the `?`, but otherwise OK!